src/structures/gdt.rs

```diff
@@ -105,6 +105,25 @@ impl GlobalDescriptorTable {
         }
     }
 
+    /// Forms a GDT from a raw table and a next_free counter.
+    ///
+    /// # Safety
+    ///
+    /// The user must ensure that the provided table contains well formed entries
+    /// and that the value of `next_free` does not exceed the maximum size of the table.
+    #[inline]
+    pub const unsafe fn from_raw_parts(table: [u64; 8], next_free: usize) -> GlobalDescriptorTable {
+        GlobalDescriptorTable { table, next_free }
+    }
```

Comment:
Exposing the internal `next_free` field seems a bit hacky. I think it would make more sense to have a `&[u64]` slice as input parameter instead (and then use `fill_from_slice`).

Reply:
I'm not sure what you mean with `fill_from_slice`, is this a method? feature?

Reply:
Sorry, I meant [`copy_from_slice`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.copy_from_slice). My idea is that the GDT is filled with the given slice and that `next_free` is set to the slice's length. I now see that your use case was const initialization, so using that method is not possible unfortunately. However, the following should be possible: 

```rust
const unsafe fn from_raw_slice(table: &[u64]) -> Self {
    let mut entries = [0; 8];
    let mut i = 0;
    loop {
        if i >= slice.len() {
            break;
        }
        
        entries[i] = table[i];
        
        i += 1;
    }
    GlobalDescriptorTable { table: entries, next_free: i }
}
```

Reply:
Sounds good to me! :+1: